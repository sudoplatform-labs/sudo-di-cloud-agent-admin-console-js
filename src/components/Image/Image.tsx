import React from 'react';

export type ImageName =
  | 'home'
  | 'settings'
  | 'sdca_light'
  | 'sdca_dark'
  | 'icon-did-line'
  | 'icon-connection-line'
  | 'icon-issuer-line'
  | 'icon-verifiable-credential-line'
  | 'icon-verifier-line'
  | 'icon-wallet-line';

export const Image: React.FC<{
  name: ImageName;
  className?: string;
}> = (props) => {
  return (
    <img
      className={props.className}
      srcSet={`/images/${props.name}.svg`}
      src="/images/default.png"
      alt={`${props.name} icon`}
    />
  );
};
