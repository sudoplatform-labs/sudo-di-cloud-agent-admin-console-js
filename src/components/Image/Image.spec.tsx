import { shallow } from 'enzyme';
import React from 'react';
import { Image } from './Image';

describe('Image', () => {
  it('should render correctly', () => {
    const component = shallow(<Image name="home" />);

    expect(component).toMatchSnapshot();
  });
});
