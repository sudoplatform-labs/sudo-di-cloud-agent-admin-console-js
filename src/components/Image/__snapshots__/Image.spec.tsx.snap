// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render correctly 1`] = `
<img
  alt="home icon"
  src="/images/default.png"
  srcSet="/images/home.svg"
/>
`;
